src/main/java/cubicchunks/asm/mixin/core/common/MixinWorldServer.java

```diff
@@ -72,16 +75,18 @@
 		this.isCubicWorld = true;
 
 		this.entitySpawner = new CubeWorldEntitySpawner();
-		this.chunkProvider = new ServerCubeCache(this);
-		this.generatorPipeline = new GeneratorPipeline(this);
-		this.thePlayerManager = new PlayerCubeMap(this);
-		this.chunkGc = new ChunkGc(getCubeCache(), getPlayerCubeMap());
+		ServerCubeCache chunkProvider = new ServerCubeCache(this);
+		this.chunkProvider = chunkProvider;
+		this.generatorPipeline = new GeneratorPipeline();
 		this.lightingManager = new LightingManager(this);
 
-
 		ICubicChunksWorldType type = (ICubicChunksWorldType) this.getWorldType();
 		type.registerWorldGen(this, this.generatorPipeline);
-		this.generatorPipeline.checkStages();
+		GeneratorPipeline.checkStages(this.generatorPipeline);
+		this.cubeGenerator = new WorldGenerator(this, this.generatorPipeline);
```

Comment:
The individual GeneratorStages are the processing elements (using their CubeProcessors) and the GeneratorPipeline defines their ordering. A registry sounds more like a system to retrieve objects using one of their properties. This implementation on the other hand discourages from accessing stages or CubeProcessors directly, instead it presents you the first stage and from that point on wards all the WorldGenerator does is advance to the next stage.